test/core/image_processing/fast_feature_detector.cpp

```diff
@@ -0,0 +1,88 @@
+//References- Following papers of Dr.Edward Rosten
+/*1.Fusing points and lines for high performance tracking.
+2.Machine learning for high-speed corner detection.
+3.Faster and better: A machine learning approach to corner detection*/
+
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/io/png.hpp>
+#include <boost/gil/image_processing/fast_feature_detector.hpp>
+#include <boost/assert.hpp>
+#include <boost/core/lightweight_test.hpp>
+#include <iostream>
+std::uint8_t null_matrix[] = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                              0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+//testing an image without a feature point
+void test1()
+{
+    boost::gil::gray8_view_t image1 = boost::gil::interleaved_view(
+        5, 8, reinterpret_cast<boost::gil::gray8_pixel_t*>(null_matrix), 5);
+    std::vector<boost::gil::point_t> keypoints;
+    std::vector<int> scores;
+    boost::gil::fast(image1, keypoints, scores, 20);
+    std::vector<boost::gil::point_t> expected_keypoints;
+    BOOST_ASSERT_MSG(
+        expected_keypoints.size() == keypoints.size(), "dimensions do not match for keypoints");
+}
+//testing color image
+void test2()
```

Comment:
Please, don't avoid blank lines as separator of code blocks/sections. Readability first.

Reply:
Done